src/components/Select/__tests__/Select.base-actions.test.tsx

```diff
@@ -132,4 +139,76 @@ describe('Select base actions', () => {
             await user.keyboard('[ArrowUp]');
         }
     });
+
+    test('find elements by "quick search" [flat list]', async () => {
+        const {getByTestId} = setup({options: QUICK_SEARCH_OPTIONS});
+        const user = userEvent.setup();
+        const selectControl = getByTestId(TEST_QA);
+        await user.click(selectControl);
+
+        await user.keyboard('3');
+        let selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 3');
+
+        await user.keyboard('5');
+        selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 35');
+
+        await user.keyboard('[Backspace]');
+        selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 3');
+    });
+
+    test('find elements by "quick search" [grouped list]', async () => {
+        const {getByTestId} = setup({options: GROUPED_QUICK_SEARCH_OPTIONS});
+        const user = userEvent.setup();
+        const selectControl = getByTestId(TEST_QA);
+        await user.click(selectControl);
+
+        await user.keyboard('3');
+        let selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 3');
+
+        await user.keyboard('5');
+        selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 35');
+
+        await user.keyboard('[Backspace]');
+        selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 3');
+    });
+
+    test('find elements by "quick search" with delay [flat list]', async () => {
+        const {getByTestId} = setup({options: QUICK_SEARCH_OPTIONS});
+        const user = userEvent.setup();
+        const selectControl = getByTestId(TEST_QA);
+        await user.click(selectControl);
+
+        await user.keyboard('3');
+        let selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 3');
+
+        await timeout(QUICK_SEARCH_TIMEOUT);
+
+        await user.keyboard('5');
+        selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 5');
+    });
+
+    test('find elements by "quick search" with delay [grouped list]', async () => {
+        const {getByTestId} = setup({options: GROUPED_QUICK_SEARCH_OPTIONS});
+        const user = userEvent.setup();
+        const selectControl = getByTestId(TEST_QA);
+        await user.click(selectControl);
+
+        await user.keyboard('3');
+        let selectedItem = getByTestId(ListQa.ACTIVE_ITEM);
+        expect(selectedItem.textContent).toBe('Value 3');
+
+        await timeout(QUICK_SEARCH_TIMEOUT);
```

Comment:
This is timeout that fires action dispatch?

Reply:
No, it is timeout for imitation of users waiting